src/reducers/wallet/NewWalletSetup.js

```diff
@@ -49,19 +56,42 @@ export default (state = INITIAL_STATE, action) => {
       return { ...state, debugMode: true };
     case actions.UPDATE_TOKEN_BALANCE:      
       const token = state.tokens[action.payload.tokenID];
-      const updatedToken = { ...token, balance: action.payload.balance }
-      const previousTokens = state.tokens;      
+      const updatedToken = { 
+        ...token, 
+        quantity: action.payload.quantity,
+        ethBalance: (action.payload.ethBalance * action.payload.quantity),
+        btcBalance: (action.payload.btcBalance * action.payload.quantity), 
+        usdBalance: (action.payload.usdBalance * action.payload.quantity), 
+        cadBalance: (action.payload.cadBalance * action.payload.quantity), 
+        eurBalance: (action.payload.eurBalance * action.payload.quantity)  
```

Comment:
@priom i think it might be a bit extra to store all the currencies...thoughts?

Reply:
@GregTheGreek @priom  the balance toggle has been added